src/value.rs

```diff
@@ -94,7 +96,26 @@ use translate::*;
 use types::{StaticType, Type};
 
 #[derive(Debug, PartialEq)]
-pub struct ValueTypeMismatch;
+pub struct ValueTypeMismatch {
```

Comment:
`ValueTypeMismatchError`. At least in the standard library all the errors end on `Error` IIRC... or not?

Reply:
I can change that, but I though it was already explicitly referring to an error, especially when used in expressions such as `Err(ValueTypeMismatch)`. `Err(ValueTypeMismatchError)` seems to state 3x there is an error :)

A [brief search in the `std` doc](https://doc.rust-lang.org/std/index.html?search=Error) shows that error types are more generic ex:

- [RecvError](https://doc.rust-lang.org/std/sync/mpsc/struct.RecvError.html)
- [Utf8Error](https://doc.rust-lang.org/std/str/struct.Utf8Error.html)
- [ParseError](https://doc.rust-lang.org/std/string/type.ParseError.html)

My first implementation was:

``` rust
pub enum ValueError {
    TypeMismatch,
}
```

but then I though it was a little too much since there was no sign of any other error types in `Value` yet, but that could be a good compromise. What do you think (of course with the `actual` and `requested` fields)? And of course, I'll change the `struct` to `ValueTypeMismatchError` if you think it is more Rust-compliant.

Reply:
Yeah I agree. I also would've done an enum first, but there's literally no other possible error with anything related to `GValue` :)

For other things there will be enum errors (like for getting properties, emitting signals, etc) as there are multiple error cases.

`TryFromIntError` would be an example from std that is a plain struct but has the `Error` suffix. That could've also be called `OutOfRange` I guess... so maybe in analogy to that, our error type should actually be called `ValueGetError` or something like that? :)

Reply:
Yes you're right, `Error`s seem to be named after the method that generated them, see in [Receiver](https://doc.rust-lang.org/std/sync/mpsc/struct.Receiver.html) which shows several examples. But they don't use the module's name. I guess this is because they are declared in the module and not re-exported in the crate root. So we could name this one `GetError` and I will no longer export it in `lib.rs`, we'll have to use `value::GetError` from other modules.

While I'm at it, I'll implement `Clone`, `Copy`, `Eq`, `Send` & `Sync` on it.

Reply:
> Edit: `Type` is neither `Send` nor `Sync`. Any reason for this?

It is both, and your error type is also both already. These are automatically implemented in this case.

Reply:
> So we could name this one `GetError` and I will no longer export it in `lib.rs`, we'll have to use `value::GetError` from other modules.

Works for me, or `ValueGetError` and re-exporting is also fine. No preference.

Reply:
> It is both, and your error type is also both already. These are automatically implemented in this case.

Indeed! Thanks